internal/handlers/alumni_handler.go

```diff
@@ -0,0 +1,95 @@
+package handlers
+
+import (
+	"backend/internal/middleware"
+	"backend/internal/models"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+type AlumniHandler struct {
+	db *gorm.DB
+}
+
+func NewAlumniHandler(db *gorm.DB) *AlumniHandler {
+	return &AlumniHandler{db: db}
+}
+
+func (h *AlumniHandler) Register(r *gin.RouterGroup) {
+	alumni := r.Group("/alumni")
+	{
+		// Public endpoints (require auth)
+		alumni.GET("", h.List)
+		alumni.GET("/:id", h.Get)
+
+		// Admin-only endpoints
+		alumni.Use(middleware.AdminRequired())
```

Comment:
I am a bit unsure, how does this work? Doesn't this get applied to the routes above? Or is it in the order of execution it is applied, meaning it is only applied to the below?